src/main/java/cms/gov/madie/terminology/controller/VsacController.java

```diff
@@ -59,7 +73,42 @@ protected String serializeFhirValueset(ValueSet fhirValueSet) {
 
   @PutMapping(path = "/validations/codes", produces = MediaType.APPLICATION_JSON_VALUE)
   public ResponseEntity<List<CqlCode>> validateCodes(
-      @RequestBody List<CqlCode> cqlCodes, @RequestParam String tgt) {
-    return ResponseEntity.ok().body(vsacService.validateCodes(cqlCodes, tgt));
+      Principal principal, @RequestBody List<CqlCode> cqlCodes) {
+    final String username = principal.getName();
+    Optional<UmlsUser> umlsUser = vsacService.findByHarpId(username);
+    if (umlsUser.isPresent() && umlsUser.get().getApiKey() != null) {
+      return ResponseEntity.ok().body(vsacService.validateCodes(cqlCodes, umlsUser.get().getTgt()));
+    }
+    return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+  }
+
+  @PostMapping(path = "/umlsLogin")
+  public ResponseEntity<String> umlsLogin(
+      Principal principal, @RequestParam(required = true, name = "apiKey") String apiKey)
+      throws InterruptedException, ExecutionException {
+    final String username = principal.getName();
+    log.debug("Entering: umlsLogin(): username = " + username);
+
+    String tgt = vsacService.getTgt(apiKey);
+    UmlsUser umlsUser = vsacService.saveUmlsUser(username, apiKey, tgt);
+    String msg = "User: " + umlsUser.getHarpId() + " successfully loggin in to UMLS.";
+    log.debug("msg = ");
+    return ResponseEntity.ok().body(msg);
+  }
+
+  @GetMapping("/checkLogin")
```

Comment:
It is preferred to use lowercase letters in URI paths, I think we can do check-login or check/login

Reply:
agreed, a URI with lowercase characters, also avoiding verbs and trying to form using nouns for resources so something like `GET /umls-credentials/status`